test(replays): test render of each individual focus area

michellewzhang opened the pull request:
Is dependent on this PR: https://github.com/getsentry/sentry/pull/54264!

Closes https://github.com/getsentry/sentry/issues/38850

michellewzhang committed:
test(replays): test render of each individual focus area

priscilawebdev committed:
fix(getting-started-docs): Fix wrong code snippet for .env (#54275)

Commit:
fix: Don't url decode abs_path before processing (#54272)

RaduW committed:
feat: Implement organization metrics estimation endpoint (#54169)

RaduW committed:
feat: Use organization metrics estimation endpoint (#54171)

rjo100 committed:
fix(crons): Fix deadlock on monitor consumer (#54140)

Move the lock to work in concert with the atomic transaction so we don't
end up with deadlocks

priscilawebdev committed:
ref(getting-started-docs): Remove source maps wizard step from node.js language and frameworks (#54282)

closes https://github.com/getsentry/sentry/issues/54281

GabeVillalobos committed:
test(hybrid-cloud): Adds silo dectorator inheritance checks (#54201)

Adds a validation when decorating test classes that prevents decorating
multiple tests with a silo decorator in the same inheritance tree, which
can cause test failures in unexpected ways.

roggenkemper committed:
feat(actionable-items): Create new actionable items endpoint (#53974)

this pr creates a new endpoint that will be used to surface all
actionable items to users on the issue details page. As of now, it will
include our sourcemap debugging errors and event errors. it combines
these errors and checks to see if any of them have been dismissed. the
response from this endpoint will be used on the front end for a new
alerting widget.

Closes https://github.com/getsentry/sentry/issues/53167

asottile-sentry committed:
ref: upgrade sentry-relay to get typed support (#54287)

mypy diff:

```diff
--- /dev/fd/63	2023-08-07 12:22:06
+++ /dev/fd/62	2023-08-07 12:22:06
-Found 5664 errors in 985 files (checked 5035 source files)
+Found 5663 errors in 984 files (checked 5035 source files)
-src/sentry/utils/outcomes.py:73: error: Cannot use isinstance() with Any type  [misc]
```

rjo100 committed:
ref(crons): Clean up monitor consumer logging (#54285)

Changes monitor consumer logging to use fstrings.

ryan953 commented on tests/acceptance/test_replay_detail.py:
```diff
@@ -83,3 +83,57 @@ def test_simple(self):
+            self.browser.wait_until_not('[data-test-id="loading-indicator"]')
+            self.browser.wait_until_not('[data-test-id="loading-placeholder"]')
+            self.browser.click('[data-test-id="Trace"]')
+            self.browser.wait_until_test_id("feature-disabled-message")
```
If you want to see the trace tab itself, instead of the disabled message, then use `with self.feature(['organizations:performance-view']);` or concat that 2nd feature with the original list.

It could matter more if there were multiple tabs that could render with 'feature-disabled-message' and we wanted to distinguish between them, but in this case it's fine either way imo.

I found that feature name by looking at the wrapper for the trace view here: https://github.com/getsentry/sentry/blob/8803d531e39ba14d4d2b643ea5faea0e30351bdc/static/app/views/replays/detail/trace/index.tsx#L13

Commit:
build(most-helpful-event): Add backend key for popup guide (#54295)

Backend for https://github.com/getsentry/sentry/issues/53915

ryan953 committed:
fix(replay): Fix replay TimestampButton alignment (#54302)

Checked all the places, things all seem to line up well now:

| Where | Pic |
| --- | --- |
| Console | <img width="302" alt="console"
src="https://github.com/getsentry/sentry/assets/187460/c43484c3-81a9-4690-a3e3-c4183f8bbcf0">
|
| Dom | <img width="125" alt="errors"
src="https://github.com/getsentry/sentry/assets/187460/ff206730-bea3-46c0-8b5b-450015c315dc">
|
| Errors | <img width="217" alt="dom"
src="https://github.com/getsentry/sentry/assets/187460/4aae13e2-b8c3-4c0a-81ad-021570146f7f">
|
| Network | <img width="188" alt="network"
src="https://github.com/getsentry/sentry/assets/187460/11f493b6-2d73-4f53-b42b-c36e52447632">
|
| Fetch/XHR panel | <img width="228" alt="network panel"
src="https://github.com/getsentry/sentry/assets/187460/3f2e3625-5179-43a1-9a12-87b2bda78f0f">
|



Note that the Network & Errors sections are both using the virtualized
grid, so same css wrapping it
And the Console & Dom sections use the virtualized list

asottile-sentry committed:
ref: fix devenv tools tests in ci (#54289)

tools intentionally doens't have access to sentry deps

lobsterkatie committed:
fix(ingest): Stop clobbering existing event metadata (#54051)

During ingest, while saving an event, we use `materialize_metadata` to gather metadata to add to the event, both as top-level properties like `title` and `culprit` and as information in the event's `metadata` property. The data it returns is then assigned to properties on the event. The intention is not to overwrite anything - it's just assumed that none of the returned data is yet there. If there is data already there, though, specifically in the `metadata` dictionary, it's lost. This fixes that by merging the existing entries in `metadata` with the incoming ones.

Note: This required changing the type of one of the parameters in `EventManager`'s version of `materialize_metadata` from `Mapping` to `dict`. In the two spots we use the function, that argument is, in fact, a dictionary, so this should be a safe change.

michellewzhang committed:
test(replays): add data-test-ids for acceptance tests (#54264)

Prepping for adding acceptance tests (added in this PR:
https://github.com/getsentry/sentry/pull/54263)

Relates to issue https://github.com/getsentry/sentry/issues/38850

michellewzhang committed:
update trace tab test
